src/plugins/telemetryTable/collections/SortedTableRowCollection.js

```diff
@@ -227,8 +245,22 @@ define(
                 this.emit('remove', removed);
             }
 
+            removeRowsFor(telemetryData, keystring) {
```

Comment:
This was because Telemetry Collections emit removed telemetry, not rows, so need to figure out another way to determine which rows to remove... still trying to figure out the best way to do it, I have a way currently, just not sure if it's foolproof.